### PR TITLE
release: fix tarball signing and Debian compression

### PR DESCRIPTION
**Pull request by ldennington**
While we added PGP signatures for tarballs in 7baac73, we did not notice
that, while ESRP returns a file with the tar.gz extension, it is actually
the signature file, not the tarball itself. Correct with this change and
validate tarball moving forward so it doesn't happen again!

Additionally, a user reported in #997 that the latest GCM Debian package
didn't work on Debian distributions. It appears that the version of dpkg
that ships with Debian [does not support zstd compression](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=892664). Enforcing xz
compression resolves the issue.
    
Finally, this provided an opportunity to clean up some unused variables in
pack.sh for Linux and to ensure we check the architecture is found before
attemping to use the ARCH variable.

These changes were validated with [this successful test run](https://github.com/ldennington/git-credential-manager/actions/runs/3795232469) in my fork.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -423,6 +427,12 @@ jobs:
       run: |
         python .github/run_esrp_signing.py deb $env:LINUX_KEY_CODE $env:LINUX_OP_CODE
         python .github/run_esrp_signing.py tar $env:LINUX_KEY_CODE $env:LINUX_OP_CODE
+    
+    - name: Re-name tarball signature file
+      shell: bash
+      run: |
+        signaturepath=$(find signed/*.tar.gz)
+        mv "$signaturepath" "${signaturepath%.tar.gz}.asc"
```
Comment (mjcheetham) [inline .github/workflows/release.yml:435]: My only question is should we not also upload/save the signature too?

**Review by mjcheetham — APPROVED** [review]
LGTM

File: .github/workflows/release.yml (hunk 1 above)
Comment (ldennington) [reply .github/workflows/release.yml:435]: Do you mean upload to the release? If so, we're already doing that! Here's the dummy release from my test run:

https://github.com/ldennington/git-credential-manager/releases/tag/untagged-4129f63ec90f2c23944a

Comment (mjcheetham) [reply .github/workflows/release.yml:435]: Oh! If we're already uploading the signature then that's great!

**Pull request merged**